mapper/tedge_mapper/src/collectd_mapper/batcher.rs

```diff
@@ -39,22 +40,22 @@ impl MessageBatch {
         &mut self,
         collectd_message: CollectdMessage,
     ) -> Result<(), DeviceMonitorError> {
-        self.message_grouper.measurement(
-            Some(collectd_message.metric_group_key),
-            collectd_message.metric_key,
-            collectd_message.metric_value,
-        )?;
+        self.message_grouper
+            .visit_start_group(&collectd_message.metric_group_key)?;
+        self.message_grouper
+            .visit_measurement(collectd_message.metric_key, collectd_message.metric_value)?;
+        self.message_grouper.visit_end_group()?;
```

Comment:
This might be simplified by adding an helper method on the `MeasurementVisitor`.

This help method would have the same signature of the former `FlatMeasurementVisitor.measurement()` and performing this `star_group; visit_measurement; end_group` sequence.

Reply:
Yes, an extension trait!